File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.js
function solution(n)
{
    let m = n;
    if(n%2!=0) n=n-1;
    let cnt=0;
    while(n!=0){
        while(n%2==0){
        n=n/2;
        }
        n-=1;
        cnt++;
    }
    
    return m%2==0? cnt: cnt+1;
    
}

File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.cpp
#include <iostream>
using namespace std;

int solution(int n)
{
    int m = n;
    if(n%2!=0) n-=1;
    int ans = 0;
    while(n!=0){
        while(n%2==0){
            n/=2;
        }
        n-=1;
        ans++;
    }

    return (m%2==1)? ans+1: ans;
}